Add support for bootstrap containers via API settings (#1387)
bottlerocket-os/bottlerocket: sources/api/README.md

This usage is scoped to the keys that have changed, updating relevant config files and restarting affected services.
See [thar-be-settings](thar-be-settings/) docs.

### Default systemd target and init workflow

> Bottlerocket uses a custom default systemd target called `preconfigured`, which is used to initialize all the services used by the API system (i. e. [settings-applier](#settings-applier)).

Reviewer: ```suggestion
Bottlerocket uses a custom default systemd target called `preconfigured`, which is used to initialize all the services used by the API system (i.e. [settings-applier](#settings-applier)).
```

Author: I removed this part of the documentation

> Bottlerocket uses a custom default systemd target called `preconfigured`, which is used to initialize all the services used by the API system (i. e. [settings-applier](#settings-applier)).

Reviewer 2: I prefer that we use the full unit name when referring to systemd units in docs (and everywhere, really):

```suggestion
Bottlerocket configures a non-standard default systemd target, `preconfigured.target`, that coordinates initialization of the host and its services before starting the API system (i. e. [settings-applier](#settings-applier)).
```

Author: I removed this change from the commits and I'll fix the documentation with #1448

> Once the target is reached, a systemd unit isolates the `configured` target, which represents the point at which the system is fully configured.

Reviewer 2: I'm not sure saying systemd "isolates" the unit in this context is clear enough. When systemd [`isolate`s to a given unit, systemd takes steps to take the system to a place where only the units that the _isolated unit_ defines (indirectly and directly) are active and inactive](https://www.freedesktop.org/software/systemd/man/systemctl.html#isolate%20UNIT) appropriately.  

```suggestion
Once `preconfigured.target` is reached, the system is isolated to the `configured.target` unit, which represents the point at which the system is fully configured.
```

Author: I removed this from the commits and I'm going to fix the documentation with #1448

Applications can depend on this in their service definition.
Services like Kubernetes and containerd depend on this.

> Upon completion, another systemd unit isolates the `multi-user` target, which is used to mark a successful boot.

Reviewer 2: ```suggestion
After reaching `configured.target`, the system then isolates itself to `multi-user.target` which indicates that the system has fully booted.
At this point, when `multi-user.target` is activated, the system records itself as having completed a successful boot.
```

Author: Oh, this is no longer true, we now mark the boot as successful on the `preconfigured` target, after `migrator.service` runs. Should I call that out here?

Author: I removed this comment, and I'm going to fix this documentation with #1448

This usage is scoped to the keys that have changed, updating relevant config files and restarting affected services.
See [thar-be-settings](thar-be-settings/) docs.

### Default systemd target and init workflow

> Bottlerocket uses a custom default systemd target called `preconfigured`, which is used to initialize all the services used by the API system (i. e. [settings-applier](#settings-applier)). Once the target is reached, a systemd unit isolates the `configured` target, which represents the point at which the system is fully configured. Applications can depend on this in their service definition. Services like Kubernetes and containerd depend on this. Upon completion, another systemd unit isolates the `multi-user` target, which is used to mark a successful boot.

Reviewer 3: Our convention is typically to write markdown docs with one sentence per line.  If there aren't any blank lines between the sentences, they are formatted as a paragraph.  Doing it this way makes it a bit easier to read/edit.  For example:
```
this is a sentence.
this is another sentence.

this will be a new paragraph.
with another sentence.
```
